Octokit.Tests.Integration/Clients/ReferencesClientTests.cs

```diff
@@ -63,6 +63,61 @@ public ReferencesClientTests()
         Assert.NotEmpty(list);
     }
 
+    [IntegrationTest]
+    public async Task ReturnsCorrectCountOfReferencesWithStart()
+    {
+        var options = new ApiOptions
+        {
+            PageSize = 5,
+            PageCount = 1,
+            StartPage = 2
+        };
+
+        var references = await _fixture.GetAll("octokit", "octokit.net", options);
+
+        Assert.Equal(5, references.Count);
+    }
+
+    [IntegrationTest]
+    public async Task ReturnsCorrectCountOfReferencesWithoutStart()
+    {
+        var options = new ApiOptions
+        {
+            PageSize = 5,
+            PageCount = 1
+        };
+
+        var references = await _fixture.GetAll("octokit", "octokit.net", options);
+
+        Assert.Equal(5, references.Count);
+    }
+
+    [IntegrationTest]
+    public async Task ReturnsDistinctReferencesBasedOnStartPage()
+    {
+        var startOptions = new ApiOptions
+        {
+            PageSize = 5,
+            PageCount = 1
+        };
+
+        var skipStartOptions = new ApiOptions
+        {
+            PageSize = 5,
+            PageCount = 1,
+            StartPage = 2
+        };
+
+        var firstRefsPage = await _fixture.GetAll("octokit", "octokit.net", startOptions);
+        var secondRefsPage = await _fixture.GetAll("octokit", "octokit.net", skipStartOptions);
+
+        foreach (var r1 in firstRefsPage)
```

Comment:
The nested loops are a bit clunky, is there a more succinct way to express this?  

eg `Assert.False(firstRefsPage.Any(x => secondRefsPage.Contains(x)));`

Reply:
Changed it.